### PR TITLE
Interactive bar animation and scrolling improvements

### PR DESCRIPTION
**Pull request by bwaresiak**
Task/Issue URL:
https://app.asana.com/0/856498667320406/893432868851070
https://app.asana.com/0/856498667320406/1126337473600500

**Description**:
- Interactive animation of top/bottom bar based on the scrolling progress.
- Tweaks to long press & bottom tap gesture recognizers to minimize impact on scroll responsiveness.

**Steps to test this PR**:

Following scenarios should work:
- Hiding bars when scrolling downwards.
- Revealing bars when scrolling above top of the view.
- Revealing bars when decelerating toward the top.
- Revealing bars when tapping on the bottom area of the screen.
- Revealing bars when tapping on top of the screen.
- Revealing bars on navigation.
- Hiding/showing bars when zooming in/out.
- Hiding disabled when web page viewport is smaller than screen (e.g: ddg lite).
- 'Tiny' swipes should work.
- Long press contextual menu should work.

---
###### Internal references:
[Software Engineering Expectations](https://app.asana.com/0/59792373528535/199064865822552)
[Technical Design Template](https://app.asana.com/0/59792373528535/184709971311943)

### CODE REVIEW

File: DuckDuckGo/BrowserChromeManager.swift [hunk 1]
```diff
@@ -19,15 +19,143 @@
...
+        
+        let cumulativeDistance = (barsHeight * transitionProgress) + distance
+        let normalizedDistance = max(cumulativeDistance, 0)
+        
+        return min(normalizedDistance / barsHeight, 1.0)
+    }
+    
+    func didScroll(in scrollView: UIScrollView) {
+
+        switch barsState {
```
Comment (brindy) [inline DuckDuckGo/BrowserChromeManager.swift]: Each of these cases could be private functions to reduce the overall function size and improve readability.

File: DuckDuckGo/BrowserChromeManager.swift [hunk 2]
```diff
@@ -19,15 +19,143 @@
 
 import UIKit
 
+private class BarsAnimator {
```
Comment (brindy) [inline DuckDuckGo/BrowserChromeManager.swift]: Maybe move this to the bottom?  I personally feel we should try and keep the main class of the file as near to the top as possible if we can.

**Review by brindy — COMMENTED** [review]
Good job! Couple of small comments for you.

**Comment by bwaresiak** [comment]
Thanks @brindy, I've made some tweaks following your comments.

**Review by brindy — APPROVED** [review]
LGTM! :shipit:

**Pull request merged**